x-pack/plugins/security_solution/public/plugin_services.ts

```diff
@@ -119,11 +119,11 @@ export class PluginServices {
       apm,
       configSettings: this.configSettings,
       savedObjectsTagging: savedObjectsTaggingOss.getTaggingApi(),
-      setHeaderActionMenu: params.setHeaderActionMenu,
+      ...(params?.setHeaderActionMenu ? { setHeaderActionMenu: params.setHeaderActionMenu } : {}),
```

Comment:
I think what I was thinking was to keep `setHeaderActionMenu` and `onAppLeave` mandatory, but just setting them as a `noop` if not available in params. This helps prevent pushing the conditional checks downstream. This works too though, so no biggie either way 🙂